### ISSUE TITLE
Project level 6 confused

### ISSUE THREAD

**Issue by jlee1007** [nus-cs2030/2122-s2#352]
Hi all for project level 6 we are suppose to allow for >1 customers to queue.
all the methods I've tried seem to require me to invoke .get() even way before the earlier customers have been served which spoils the timing for them too.

does anyone know how to circumvent this problem? or what should be the correct way to update the nextAvailTime of the server?

thank you!

**Comment by wk1267** [comment]
HI, u shd only call .get once, which shd be when u r serving them

**Comment by YangHaoying** [comment]
Hi! You should stick to principles in level 5 which is only to call .get() once for every customer in their ServeEvent. An approach would be to only return actual ServeEvent for a waiting customer if it is at the top of their queue, and some PendingEvent class if not. Then you should need to get the service time only once, as now the customer only has one actual ServeEvent.

**Comment by yjjjiajie** [comment]
for me, i invoked the .get() in my simulate, which ensures that I only get the timing when i rly need to serve

**Comment by jlee1007** [comment]
@YangHaoying thanks! how did u update the timing for the pending event tho

**Comment by loqir** [comment]
> @YangHaoying thanks! how did u update the timing for the pending event tho

yeah i was thinking the same thing too! i wonder if it's possible to track the customer that the server is currently serving and then add this servicetime to the current time to update the timing for the pending event, maybe through an attribute in the Server class

**Comment by abigailxj** [comment]
@loqir hello, how did you update the servers for the pending events such that the service times are updated? since pending events are created before the first person in queue gets served

**Comment by loqir** [comment]
@abigailxj hello! in my pending event, i did not update the server. i first checked if the customer was the first in the waiting list. if so, i will return a serve event. else, i will return another pending event which event time included the service time of the server.
